cg/store/filters/status_illumina_metrics_filters.py

```diff
@@ -0,0 +1,53 @@
+"""Filters for the Illumina Sample Sequencing Metrics objects."""
+
+from enum import Enum
+from typing import Callable
+
+from sqlalchemy.orm import Query
+
+from cg.store.models import (
+    IlluminaFlowCell,
+    IlluminaSampleSequencingMetrics,
+    IlluminaSequencingRun,
+    Sample,
+)
+
+
+def filter_by_run_id(metrics: Query, run_id: str, **kwargs) -> Query:
+    """Filter metrics by run id."""
+    joined_query: Query = metrics.join(IlluminaSequencingRun).join(IlluminaFlowCell)
+    return joined_query.filter(IlluminaFlowCell.internal_id == run_id)
+
+
+def filter_by_sample_internal_id(metrics: Query, sample_internal_id: str, **kwargs) -> Query:
+    """Filter metrics by sample internal id."""
+    joined_query: Query = metrics.join(Sample)
+    return joined_query.filter(Sample.internal_id == sample_internal_id)
```

Comment:
This is not what I meant. The expensive joins still happen at this point. since you first join full tables before filtering any.

Reply:
fixed now!